ngi_pipeline/engines/qc_ngi/workflows.py

```diff
@@ -153,53 +172,58 @@ def workflow_fastq_screen(input_files, output_dir, config):
     subsample_reads = config.get("qc", {}).get("fastq_screen", {}).get("subsample_reads")
 
     # Determine which files need processing
+    fastq_files = flatten(input_files) # Fastq_screen cares not for your "read pairs" anymore from version 1.5
+    # Verify that we in fact need to run this on these files
     fastq_screen_output_file_tmpls = ("{}_screen.png", "{}_screen.txt")
     fastq_to_analyze = set()
-    for elt in input_files:
-        # This may be a read pair
-        if type(elt) is list:
-            # Changing list to tuple so we can use it in the set() (lists aren't hashable)
-            elt = tuple(elt)
-            # fastq_screen uses the name of the first read of the pair for output files
-            fastq_file = elt[0]
+    for fastq_file in fastq_files:
+        m = re.match(r'([\w-]+).fastq', os.path.basename(fastq_file))
```

Comment:
this code looks very similar to the code above.. I think a refactoring to remove the redundancy would be helpful